frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1298,6 +1369,12 @@ function ReaderHighlight:onReadSettings(config)
         disable_highlight = G_reader_settings:readSetting("highlight_disabled") or false
     end
     self.view.highlight.disabled = disable_highlight
+
+    -- panel zoom settings
+    self.panel_zoom_enabled = config:readSetting("panel_zoom_enabled")
```

Comment:
I would wrap that too (loading & saving) into a `if self.document.info.has_pages then`.
(I can't update this setting as the menu is not displayed on CRE document, but if I were to toggle manually the saved setting in a metadata.epub.lua, things could break - so, I feel it's better to have explicite checks.)

Reply:
Good catch, since it's toggable via gesture